pkg/util/set.go

```diff
@@ -62,6 +62,17 @@ func (s StringSet) HasAll(items ...string) bool {
 	return true
 }
 
+// Difference returns a set of objects that are not in s2
+func (s StringSet) Difference(s2 StringSet) StringSet {
```

Comment:
For non mathematicians (me) who had to think hard about this, can you make the comment a bit more explicit?  Maybe a small example?